From 80f390eb2590f0ed4bb7a40f10b331b2e069df6c Mon Sep 17 00:00:00 2001
From: Juan Gutierrez <juan.gutierrez@nxp.com>
Date: Mon, 26 Sep 2016 14:31:19 -0500
Subject: [PATCH 1253/5242] MLK-13284 ARM: dts: imx: generic support for
 6sxscm 512mb evb

commit  8ac10d03992fde4e94bd6beaa2e96f4b162ccef4 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add the generic dtsi configuration support for the i.MX6SX SCM
platforms with 512MB of DDR memory mapping.

Signed-off-by: Juan Gutierrez <juan.gutierrez@nxp.com>
Signed-off-by: Alejandro Sierra <alejandro.sierra@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sxscm-512mb-evb-ldo.dts    |   10 ++++++++++
 arch/arm/boot/dts/imx6sxscm-512mb-evb-m4-ldo.dts |   18 ++++++++++++++++++
 arch/arm/boot/dts/imx6sxscm-512mb.dtsi           |   14 ++++++++++++++
 3 files changed, 42 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx6sxscm-512mb-evb-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6sxscm-512mb-evb-m4-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6sxscm-512mb.dtsi

diff --git a/arch/arm/boot/dts/imx6sxscm-512mb-evb-ldo.dts b/arch/arm/boot/dts/imx6sxscm-512mb-evb-ldo.dts
new file mode 100644
index 0000000..f34b358
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sxscm-512mb-evb-ldo.dts
@@ -0,0 +1,10 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sxscm-1gb-evb-ldo.dts"
+#include "imx6sxscm-512mb.dtsi"
diff --git a/arch/arm/boot/dts/imx6sxscm-512mb-evb-m4-ldo.dts b/arch/arm/boot/dts/imx6sxscm-512mb-evb-m4-ldo.dts
new file mode 100644
index 0000000..a43f8d7
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sxscm-512mb-evb-m4-ldo.dts
@@ -0,0 +1,18 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sxscm-1gb-evb-m4-ldo.dts"
+/{
+	memory {
+		linux,usable-memory = <0x80000000 0x1ff00000> ;
+	};
+};
+
+&rpmsg{
+	reg = <0x9FFF0000 0x8000>;
+};
diff --git a/arch/arm/boot/dts/imx6sxscm-512mb.dtsi b/arch/arm/boot/dts/imx6sxscm-512mb.dtsi
new file mode 100644
index 0000000..023c997
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sxscm-512mb.dtsi
@@ -0,0 +1,14 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+/ {
+	memory {
+		reg = <0x80000000 0x20000000>;
+	};
+
+};
-- 
1.7.9.5

